xarray/core/dataset.py

```diff
@@ -2905,7 +2906,7 @@ def reduce(self, func, dim=None, keep_attrs=None, numeric_only=False,
         else:
             dims = set(dim)
 
-        missing_dimensions = [dim for dim in dims if dim not in self.dims]
+        missing_dimensions = [dim for d in dims if d not in self.dims]
```

Comment:
Here and below: This looks like a mistake -- `dim` is no longer the loop variable.

Reply:
Thanks!